src/vsphere_cpi/spec/unit/cloud/vsphere/nsxt_provider_spec.rb

```diff
@@ -633,4 +634,43 @@
       nsxt_provider.remove_vm_from_server_pools(vm_ip_address)
     end
   end
+
+  describe '#retrieve_all_ns_groups_with_pagination' do
+    let(:nsgroup_1) do
+      NSXT::NSGroup.new(:id => 'id-1', :display_name => 'test-nsgroup-1')
+    end
+    let(:nsgroup_2) do
+      NSXT::NSGroup.new(:id => 'id-2', :display_name => 'test-nsgroup-2')
+    end
+    let(:nsgroup_3) do
+      NSXT::NSGroup.new(:id => 'id-3', :display_name => 'test-nsgroup-3')
+    end
+    let(:nsgroup_4) do
+      NSXT::NSGroup.new(:id => 'id-4', :display_name => 'test-nsgroup-4')
+    end
+
```

Comment:
Can you add a test where the result list is empty? pages == 0

Reply:
Done